tests/rados/test_mon_system_operations.py

```diff
@@ -0,0 +1,268 @@
+        )
+        raise Exception("Mon not in quorum error")
+
+    # Writing Objects on pool post mon Addition and OSD down
+    rados_obj.bench_write(pool_name=pool_name, byte_size=1024, rados_write_duration=50)
+
+    if not rados_obj.change_osd_state(action="start", target=osd_id):
+        log.error(f"Could not start OSD : {osd_id}")
+        raise Exception("OSD not started error")
+
+    # Now that the OSD is restarted, waiting for the recovery while the mon is Added
+    log.debug(
+        " waiting for PGs to settle down and recover post OSD stop and start during mon Addition"
+    )
+    method_should_succeed(wait_for_clean_pg_sets, rados_obj, 5000)
+
+    if not rados_obj.run_pool_sanity_check():
+        log.error("Checks failed post mon removal and OSD rebalance")
+        raise Exception("Secnario 4 of Mon tests failed")
+
+    log.info("Completed scenario 4. Pass")
+
+    log.info(
+        "---------- starting workflow 5 - Start and Stop of mon daemons with IOs --------"
+    )
+    test_mon_host = random.choice(init_mon_nodes)
+    if not rados_obj.change_daemon_systemctl_state(
+        action="stop", daemon_type="mon", daemon_id=test_mon_host.hostname
+    ):
+        log.error(f"Failed to stop mon on host {test_mon_host.hostname}")
+        raise Exception("Mon stop failure error")
+
+    # Writing Objects on pool post mon down
+    rados_obj.bench_write(pool_name=pool_name, byte_size=1024, rados_write_duration=50)
+
+    if not rados_obj.run_pool_sanity_check():
+        log.error("Checks failed post mon down")
+        raise Exception("Sanity check failed post mon down")
+
+    if not rados_obj.change_daemon_systemctl_state(
+        action="start", daemon_type="mon", daemon_id=test_mon_host.hostname
+    ):
+        log.error(f"Failed to start mon on host {test_mon_host.hostname}")
+        raise Exception("Mon stop failure error")
+
+    # Writing Objects on pool post mon down
+    rados_obj.bench_write(pool_name=pool_name, byte_size=1024, rados_write_duration=50)
+
+    if not rados_obj.run_pool_sanity_check():
+        log.error("Checks failed post mon down")
+        raise Exception("Sanity check failed post mon start")
+
+    log.info("Completed scenario 5. Pass")
+
+    log.info(
+        "---------- starting workflow 6 - rolling reboot of all mon daemons with IOs --------"
+    )
+
+    for mon_node in init_mon_nodes:
+        if not rados_obj.change_daemon_systemctl_state(
+            action="restart", daemon_type="mon", daemon_id=mon_node.hostname
+        ):
+            log.error(f"Failed to restart mon on host {test_mon_host.hostname}")
+            raise Exception("Mon restart failure error")
+
+        # Writing Objects on pool post mon restart
+        rados_obj.bench_write(
+            pool_name=pool_name, byte_size=1024, rados_write_duration=50
+        )
+
+        if not rados_obj.run_pool_sanity_check():
+            log.error("Checks failed post mon down")
```

Comment:
please modify to "Checks failed post mon restart" to reflect the scenario

Reply:
Updated log lines.